frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -573,35 +572,54 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
         String releaseId = request.getParameter(PortalConstants.RELEASE_ID);
         String attachmentContentId = request.getParameter(PortalConstants.ATTACHMENT_ID);
         String attachmentName = request.getParameter(PortalConstants.ATTACHMENT_NAME);
+        Map<String, Set<String>> licenseToSrcFilesMap = new LinkedHashMap<>();
         boolean includeConcludedLicense = new Boolean(request.getParameter(PortalConstants.INCLUDE_CONCLUDED_LICENSE));
 
         ComponentService.Iface componentClient = thriftClients.makeComponentClient();
         LicenseInfoService.Iface licenseInfoClient = thriftClients.makeLicenseInfoClient();
 
-        Set<String> concludedLicenseIds = new HashSet<>();
-        Set<String> mainLicenseNames = new HashSet<String>();
-        Set<String> otherLicenseNames = new HashSet<String>();
+        final Set<String> concludedLicenseIds = new TreeSet<String>();
+        Set<String> mainLicenseNames = new TreeSet<String>();
+        Set<String> otherLicenseNames = new TreeSet<String>();
+        AttachmentType attachmentType = AttachmentType.OTHER;
+        long totalFileCount = 0;
         try {
             Release release = componentClient.getReleaseById(releaseId, user);
             List<LicenseInfoParsingResult> licenseInfoResult = licenseInfoClient.getLicenseInfoForAttachment(release,
                     attachmentContentId, includeConcludedLicense, user);
+            attachmentType = release.getAttachments().stream().filter(att -> attachmentContentId.equals(att.getAttachmentContentId())).map(Attachment::getAttachmentType).findFirst().orElse(null);
             List<LicenseNameWithText> licenseWithTexts = licenseInfoResult.stream()
                     .flatMap(result -> result.getLicenseInfo().getLicenseNamesWithTexts().stream())
                     .filter(license -> !license.getLicenseName().equalsIgnoreCase(SW360Constants.LICENSE_NAME_UNKNOWN)
                             && !license.getLicenseName().equalsIgnoreCase(SW360Constants.NA)
                             && !license.getLicenseName().equalsIgnoreCase(SW360Constants.NO_ASSERTION)) // exclude unknown, n/a and noassertion
                     .collect(Collectors.toList());
-            if (attachmentName.endsWith(".rdf")) {
-                concludedLicenseIds = licenseInfoResult.stream()
-                        .flatMap(singleResult -> singleResult.getLicenseInfo().getConcludedLicenseIds().stream())
-                        .collect(Collectors.toSet());
-                otherLicenseNames = licenseWithTexts.stream().map(LicenseNameWithText::getLicenseName).collect(Collectors.toSet());
+
+            if (attachmentName.endsWith(PortalConstants.RDF_FILE_EXTENSION)) {
+                if (AttachmentType.INITIAL_SCAN_REPORT.equals(attachmentType)) {
+                    totalFileCount = licenseWithTexts.stream().map(LicenseNameWithText::getSourceFiles).filter(src -> src != null).mapToInt(Set::size).sum();
```

Comment:
`Objects::nonNull` will look good may be

Reply:
Added `Objects::nonNull`